src/convertToObject.js

```diff
@@ -5,8 +5,26 @@
  *
  * @return {object}
  */
-function convertToObject(sourceString) {
-  // write your code here
+function convertToObject(stylesString) {
+  return stylesString
+    .split(';')
+    .map((style) => style.trim())
+    .filter(Boolean)
+    .map((style) => {
+      const [property, value] = style.split(':');
+
+      if (property === undefined || value === undefined) {
+        return null;
+      }
+
+      return [property.trim(), value.trim()];
+    })
+    .filter(Boolean)
+    .reduce((obj, [property, value]) => {
+      obj[property] = value;
+
+      return obj;
```

Comment:
[CODE KNOWLEDGE] - Don't use iteration methods to modify an already existing array/object. In this case, you're directly modifying the `obj` parameter inside the `reduce` callback. This is not a good practice as it can lead to side effects. Instead, you should return a new object with the accumulated properties.